### PR TITLE
fix(ESSNTL-4617): Don't remove HostGroups in PATCH if they're re-added

### PR DESCRIPTION
**Pull request by kruai**
# Overview

This PR is being created to address an issue discovered for [ESSNTL-4617](https://issues.redhat.com/browse/ESSNTL-4617). It makes it so that in `PATCH /groups` requests, if the Host ID list in the payload contains hosts that are already associated with the group, these hosts only produce one MQ event. Before this PR, it would produce a message when the host was removed from the group, and one when the host is re-added.

## PR Checklist

- [x] Keep PR title short, ideally under 72 characters
- [ ] Descriptive comments provided in complex code blocks
- [x] Tests: validate optimal/expected output
- [ ] Tests: validate exceptions and failure scenarios
- [ ] Tests: edge cases
- [ ] Recovers or fails gracefully during potential resource outages (e.g. DB, Kafka)
- [x] Uses [type hinting](https://docs.python.org/3/library/typing.html), if convenient
- [ ] Documentation, if this PR changes the way other services interact with host inventory
- [ ] Links to related PRs

## Secure Coding Practices Documentation Reference

You can find documentation on this checklist [here](https://github.com/RedHatInsights/secure-coding-checklist).

## Secure Coding Checklist

- [ ] Input Validation
- [ ] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [ ] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [ ] General Coding Practices

### CODE REVIEW

**Comment by kruai** [comment]
Marked this as a draft, since Fero discovered another bug in the same Jira. I'll attempt to address that in this PR as well.

File: tests/test_api_groups_update.py [hunk 1]
```diff
@@ -232,3 +232,25 @@ def test_patch_group_name_only(
...
+    mocker.patch.object(event_producer, "write_event")
+    group = db_create_group_with_hosts("test_group", 5)
+    group_id = group.id
+    host_id_list = [str(host.id) for host in group.hosts]
+
+    patch_doc = {"host_ids": host_id_list}
+    response_status, response_data = api_patch_group(group_id, patch_doc)
+    assert_response_status(response_status, 200)
+
+    # Validate that only 2 messages were sent out, updating the group data
```
Comment (thearifismail) [inline tests/test_api_groups_update.py:250]: @asa Is this old comment? If the comment is correct, how do you verify that 2 messages were sent?

**Review by thearifismail — COMMENTED** [review]
LGTM, except one comment?

File: tests/test_api_groups_update.py (hunk 1 above)
Comment (kruai) [reply tests/test_api_groups_update.py]: Oops, that's from before I changed the number of hosts from 2 to 5. I've updated the comment

Comment (kruai) [reply tests/test_api_groups_update.py]: And as far as how I verify the number of messages sent, it's this line: `assert event_producer.write_event.call_count == 5`

Comment (thearifismail) [reply tests/test_api_groups_update.py]: Good this makes sense.

**Review by thearifismail — APPROVED** [review]
LGTM

**Review by CodeHeeler — APPROVED** [review]
LGTM

**Review by fstavela — APPROVED** [review]

**Comment by kruai** [comment]
This is obsoleted by #1306, which makes broader fixes that address this issue along with others.

**Pull request closed without merging**